### PR TITLE
feat: add `topmodel` `init_config` models

### PR DESCRIPTION
**Pull request by aaraney**
Initial structures for parsing [`topmodel`](https://github.com/noaa-owp/topmodel) configuration files. Still a few things to sort so it is more compliant with `ngen.init_config`, but this is a good start.

### CODE REVIEW

**Comment by aaraney** [comment]
@hellkite500, there are a few assumptions that ive made here that we might want to change. In either case, this is ready for a first round of review.

File: python/ngen_conf/src/ngen/config/init_config/topmodel.py [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+        assert distarea_lnaotb_str != "", "missing `dist_area_lnaotb` `lnaotb`"
+        distarea_lnaotb = distarea_lnaotb_str.split(" ")
+        assert (
+            len(distarea_lnaotb) == 2
+        ), f"incorrect number of fields in `dist_area_lnaotb` `lnaotb`, got {len(ntopo_area)}"
+        data["dist_area_lnaotb"] = float(distarea_lnaotb[0])
+        data["lnaotb"] = float(distarea_lnaotb[1])
+
+        num_channels_str = reader.readline()
+        assert distarea_lnaotb_str != "", "missing `num_channels`"
```
Comment (hellkite500) [inline python/ngen_conf/src/ngen/config/init_config/topmodel.py:129]: Wrong variable in check here, should be num channels

File: python/ngen_conf/src/ngen/config/init_config/topmodel.py [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+    input: pathlib.Path = pathlib.Path("/dev/null")
+    subcat: PathPair[TopModelSubcat]
+    params: PathPair[TopModelParams]
+    output: pathlib.Path = pathlib.Path("/dev/null")
+    hyd: pathlib.Path = pathlib.Path("/dev/null")
+
+    @validator("stand_alone", pre=True)
+    @classmethod
+    def _coerce_stand_along(cls, value: str | int):
+        return int(value)
```
Comment (hellkite500) [inline python/ngen_conf/src/ngen/config/init_config/topmodel.py:290]: Can we warn if NGEN configs are marked with the standalone flag?

File: python/ngen_conf/src/ngen/config/init_config/topmodel.py [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+        data["yes_print_output"] = int(cat_imap_out[2])
+
+        assert data["num_sub_catchments"] == 1, "only 1 subcatchment supported"
+
+        # drop trailing newline
+        title = reader.readline().rstrip("\r\n")
+        assert (
+            len(title.encode()) <= 256
+        ), "`title` must be less that 256 bytes in length"
+        data["subcat"] = title
```
[the remark this answers is not in the document]
Comment (aaraney) [reply python/ngen_conf/src/ngen/config/init_config/topmodel.py:106]: Yeah, I get what you mean. Lets meet in the middle and error if its `>255`. I would rather fail here if someone is using this for validation purposes and they are being really pedantic about the meaning of validation (never been me before 👀).

File: python/ngen_conf/src/ngen/config/init_config/topmodel.py (hunk 2 above)
Comment (aaraney) [reply python/ngen_conf/src/ngen/config/init_config/topmodel.py:290]: Yeah, I think that's fair. I'd left it this way originally as I was implementing the bare minimum parser for NextGen. I'll go back and add the functionality to support more complex configuration files.

I think it's probably worth maintaining the complexity of a more complex parser and model for this module in particular. I envision TOPMODEL might be a candidate bmi module for a layered configuration in the future. What are your thoughts, @hellkite500?

Comment (hellkite500) [reply python/ngen_conf/src/ngen/config/init_config/topmodel.py:290]: I generally agree. And I think if we are building ngen specific configuration and validation, we should at least provide warnings of incompatible instances when we can.

Comment (hellkite500) [reply python/ngen_conf/src/ngen/config/init_config/topmodel.py:364]: I don't see it 👀

Comment (aaraney) [reply python/ngen_conf/src/ngen/config/init_config/topmodel.py]: Ooooops, you right. I put a warning somewhere else but not here. Just added it. My b.

**Pull request merged**